walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,186 @@
 
 package com.google.sps;
 
+import com.google.sps.Event;
+import com.google.sps.Events;
+import com.google.sps.TimeRange;
+import com.google.sps.MeetingRequest;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Set;
 
 public final class FindMeetingQuery {
-  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
+    Collection<TimeRange> result = new ArrayList<>(); // the return value: just a list of time ranges.
+    Collection<String> attendees = request.getAttendees();
+    Collection<String> optAttendees = request.getOptionalAttendees();
+    Iterator<Event> itr = events.iterator();
+
+    //noOptionsForTooLongOfARequest()
```

Comment:
Comments in source code should be meaningful, describing what the code is doing, rather than referring to test points